builtin/providers/aws/provider.go

```diff
@@ -107,6 +107,15 @@ func Provider() terraform.ResourceProvider {
 			"aws_ami":                              resourceAwsAmi(),
 			"aws_ami_copy":                         resourceAwsAmiCopy(),
 			"aws_ami_from_instance":                resourceAwsAmiFromInstance(),
+			"aws_api_gateway":                      resourceAwsApiGateway(),
```

Comment:
Is there any reason why can't we call this resource `aws_api_gateway_rest_api` and make it match the AWS API naming convention?